frontend/pages/hosts/HostDetailsPage/SoftwareVulnerabilities/SoftwareVulnerabilities.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import FleetIcon from "components/icons/FleetIcon";
+import softwareInterface from "interfaces/software";
+
+const baseClass = "software-vulnerabilities";
+
+class SoftwareVulnerabilities extends Component {
+  static propTypes = {
+    softwareList: PropTypes.arrayOf(softwareInterface),
+  };
+
+  render() {
+    const { softwareList } = this.props;
+
+    const vulsList = [];
+
+    const vulnerabilitiesListMaker = () => {
+      softwareList.forEach((software) => {
+        if (software.vulnerabilities) {
```

Comment:
Regarding a comment above, adding `vulnerabilities` to the types interface as a required array will remove the need for a check. Also, adding the type should force a default of `[]` somewhere in the code so we're sure to always pass an empty array if nothing is given.